bob/src/hw_metrics_collector.rs

```diff
@@ -242,52 +266,83 @@ impl DiskStatCollector {
         }
 
         DiskStatCollector {
-            iostat_avl: true,
+            procfs_avl: true,
             disk_metric_data,
         }
     }
 
     fn collect_and_send_metrics(&mut self) -> Result<(), CommandError> {
-        let iostat_lines = self.collect_iostat()?;
-        for i in iostat_lines {
+        let uptime = self.uptime()?;
+        let diskstats = self.diskstats()?;
+        for i in diskstats {
             let lsp: Vec<&str> = i.split_whitespace().collect();
-            if lsp.len() == 21 {
-                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[0]) {
-                    if let (Ok(rs), Ok(ws)) = (
-                        // readops per s count is in 1st column
-                        Self::parse_iostat_result(lsp[1]),
-                        // writeops per s count is in 7th column
-                        Self::parse_iostat_result(lsp[7]),
+            if lsp.len() >= 8 {
+                if let Some(metric_prefix) = self.disk_metric_data.get(lsp[2]) {
+                    if let (Ok(r_ios), Ok(w_ios)) = (
+                        // successfull reads count is in 3rd column
+                        lsp[3].parse::<f64>(),
+                        // successfull writes count is in 7th column
+                        lsp[7].parse::<f64>(),
                     ) {
-                        let iops = rs + ws;
+                        let iops = (r_ios + w_ios) / uptime;
```

Comment:
We need the current number of iops, not the average since the system was started. The easiest way is to calculate the delta between this call and the previous one, with an additional check that the elapsed time is greater than 1 second